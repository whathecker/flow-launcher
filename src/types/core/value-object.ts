export type PriorityTier = "n/a" | "highest" | "high" | "mid" | "low";

export type PriorityMeasure = "n/a" | "yes" | "no"; // TODO: consider to use MIXED type (boolean and null) instead?

export class Priority {
  constructor(
    public tier: PriorityTier,
    public importance: PriorityMeasure,
    public urgency: PriorityMeasure,
  ) {}
}
