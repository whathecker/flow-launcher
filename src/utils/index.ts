export { default as navigationRef } from "./navigationRef";
export { default as taskFilters } from "./taskFilters";
export { default as colorRenderer } from "./colorRenderer";
export { default as labelRenderer } from "./labelRenderer";
