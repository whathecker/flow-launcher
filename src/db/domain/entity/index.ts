import { Priority } from "../value-object";

export type ID = number | string;

export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public id: ID,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
  ) {}
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
    public id?: ID,
  ) {}
}
