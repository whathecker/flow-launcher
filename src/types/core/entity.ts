import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export interface IAddTaskInput {
  title: string;
  description: string;
  goal_id: string;
}

export interface IFetchTasksInput {
  goal_id: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public status: TaskStatus,
    public description?: string,
    public priority?: Priority,
  ) {}
}
