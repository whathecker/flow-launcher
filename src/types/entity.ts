import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[] | [],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
  ) {}
}
