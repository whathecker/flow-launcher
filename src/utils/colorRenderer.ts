import { GoalColor } from "../types/core/entity";
import {
  PriorityTier,
  PriorityBucketColor,
  ColorShadeForFirstGoalColor,
  ColorShadeForSecondGoalColor,
  ColorShadeForThirdGoalColor,
  ColorShadeForFourthGoalColor,
} from "../types/core/value-object";

const assignColorForGoal = (index: number): GoalColor => {
  if (_isFirstColor(index)) return "#E9695B";

  let color;
  switch (_findColorToRender(index)) {
    case 1:
      color = "#84C5E8";
      break;
    case 2:
      color = "#1C88BA";
      break;
    case 3:
      color = "#F0B541";
      break;
    default:
      color = "#E9695B";
      break;
  }

  return color as GoalColor;
};

function _isFirstColor(index: number): boolean {
  if (index === 0 || index % 4 === 0) return true;

  return false;
}

function _findColorToRender(index: number): number {
  return index % 4;
}

const getColorForPrioBucket = (
  goalColor: GoalColor,
  prio: PriorityTier,
): PriorityBucketColor | "#FEFEF8" => {
  let color;
  switch (goalColor) {
    case "#E9695B":
      color = _findColorShadeForFirstColor(prio);
      break;
    case "#84C5E8":
      color = _findColorShadeForSecondColor(prio);
      break;
    case "#1C88BA":
      color = _findColorShadeForThirdColor(prio);
      break;
    case "#F0B541":
      color = _findColorShadeForFourthColor(prio);
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as PriorityBucketColor;
};

function _findColorShadeForFirstColor(
  prio: PriorityTier,
): ColorShadeForFirstGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#C2571A";
      break;
    case "high":
      color = "#DA621E";
      break;
    case "mid":
      color = "#F26D21";
      break;
    case "low":
      color = "#F58B4C";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForFirstGoalColor;
}

function _findColorShadeForSecondColor(
  prio: PriorityTier,
): ColorShadeForSecondGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#107896";
      break;
    case "high":
      color = "#0D9BC3";
      break;
    case "mid":
      color = "#2FAFD3";
      break;
    case "low":
      color = "#65CBE9";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForSecondGoalColor;
}

function _findColorShadeForThirdColor(
  prio: PriorityTier,
): ColorShadeForThirdGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#093145";
      break;
    case "high":
      color = "#0D3D56";
      break;
    case "mid":
      color = "#0D5275";
      break;
    case "low":
      color = "#3C6478";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForThirdGoalColor;
}

function _findColorShadeForFourthColor(
  prio: PriorityTier,
): ColorShadeForFourthGoalColor | "#FEFEF8" {
  let color;
  switch (prio) {
    case "highest":
      color = "#BCA136";
      break;
    case "high":
      color = "#D3B53D";
      break;
    case "mid":
      color = "#EBC944";
      break;
    case "low":
      color = "#EFD469";
      break;
    default:
      color = "#FEFEF8";
      break;
  }
  return color as ColorShadeForFourthGoalColor;
}

export default { assignColorForGoal, getColorForPrioBucket };
