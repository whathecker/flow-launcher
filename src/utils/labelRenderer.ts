import { PriorityTier } from "../types/core/value-object";

const renderPrioBucketLabel = (prio: PriorityTier): string => {
  let label = "";
  switch (prio) {
    case "highest":
      label = "Highest priority";
      break;
    case "high":
      label = "High priority";
      break;
    case "mid":
      label = "Mid priority";
      break;
    case "low":
      label = "Low priority";
      break;
    default:
      // TODO: Add error handling here
      break;
  }
  return label;
};

export default { renderPrioBucketLabel };
