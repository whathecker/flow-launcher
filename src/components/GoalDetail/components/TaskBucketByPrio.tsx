import React from "react";
import { Image } from "react-native";
import { Touchable, View, Text } from "../../Themed";
import { Color, Container, Typography } from "../../../styles";
import { Task, GoalColor } from "../../../types/core/entity";
import { PriorityTier } from "../../../types/core/value-object";
import { colorRenderer, labelRenderer, navigationRef } from "../../../utils";

type ActiveBucketProps = {
  prio: PriorityTier;
  backgroundColor: string;
  textColor: string;
  tasks: Task[];
};

function _trimTaskTitle(input: string): string {
  if (input.length > 10) {
    return input.slice(0, 20).concat(" ...");
  } else {
    return input;
  }
}

const ActiveBucket: React.FC<ActiveBucketProps> = ({
  prio,
  tasks,
  backgroundColor,
  textColor,
}: ActiveBucketProps) => {
  return (
    <Touchable
      onPress={() => {
        navigationRef.navigate("TasksByPrio", {
          backgroundColor: backgroundColor,
          prio: prio,
        });
      }}
      style={{
        width: "100%",
        height: 120,
        backgroundColor: backgroundColor,
        marginTop: 5,
        marginBottom: "3%",
        paddingTop: "4%",
        paddingBottom: "8%",
        borderWidth: 1,
        borderColor: Color.light.defaultBorder,
        borderRadius: 5,
      }}
    >
      <View
        style={{
          backgroundColor: backgroundColor,
          paddingLeft: 15,
        }}
      >
        <Text
          style={{
            ...Typography.h4,
            fontSize: 15,
            fontWeight: "bold",
            color: textColor,
          }}
        >
          {labelRenderer.renderPrioBucketLabel(prio)}
        </Text>
      </View>

      <View
        style={{
          backgroundColor: backgroundColor,
          paddingTop: "6%",
          paddingBottom: "2%",
          paddingLeft: "10%",
        }}
      >
        <Text
          style={{
            ...Typography.p,
            color: textColor,
            fontSize: 13,
          }}
        >{`${_trimTaskTitle(tasks[0].title)}`}</Text>
        {tasks.length > 1 ? (
          <Text
            style={{
              ...Typography.p,
              color: textColor,
              fontSize: 13,
              textDecorationLine: "underline",
              paddingTop: 5,
            }}
          >{`+ ${tasks.length - 1} more`}</Text>
        ) : null}
        <View
          style={{
            backgroundColor: backgroundColor,
            position: "absolute",
            top: -18,
            right: 15,
          }}
        >
          <Image
            style={{ width: 18, height: 18 }}
            source={require(`../../../../assets/images/right-arrow-white.png`)}
          />
        </View>
      </View>
    </Touchable>
  );
};

type EmptyBucketProps = {
  prio: PriorityTier;
  backgroundColor: string;
  textColor: string;
  tasks?: Task[];
};

const EmptyBucket: React.FC<EmptyBucketProps> = ({
  prio,
  backgroundColor,
  textColor,
}: EmptyBucketProps) => {
  return (
    <View
      style={{
        width: "100%",
        height: 120,
        backgroundColor: backgroundColor,
        marginTop: 5,
        marginBottom: "3%",
        paddingTop: "4%",
        paddingBottom: "8%",
        borderWidth: 1,
        borderColor: Color.light.defaultBorder,
        borderRadius: 5,
      }}
    >
      <View
        style={{
          backgroundColor: backgroundColor,
          paddingLeft: 15,
        }}
      >
        <Text
          style={{
            ...Typography.h4,
            fontSize: 15,
            fontWeight: "bold",
            color: textColor,
          }}
        >
          {labelRenderer.renderPrioBucketLabel(prio)}
        </Text>
      </View>
      <View
        style={{
          ...Container.centerAligned,
          backgroundColor: backgroundColor,
          paddingTop: "9.5%",
        }}
      >
        <Text
          style={{
            ...Typography.p,
            fontSize: 12,
            color: textColor,
          }}
        >{`No tasks in this list`}</Text>
      </View>
    </View>
  );
};

type TaskBucketByPrioProps = {
  prio: PriorityTier;
  tasks: Task[];
  goalColor: GoalColor;
};

const TaskBucketByPrio: React.FC<TaskBucketByPrioProps> = ({
  prio,
  tasks,
  goalColor,
}: TaskBucketByPrioProps) => {
  const isTasksEmpty = tasks.length === 0;

  const backgroundColor = isTasksEmpty
    ? "#FEFEF8"
    : colorRenderer.getColorForPrioBucket(goalColor, prio);

  const textColor = isTasksEmpty
    ? Color.light.subtleLabel
    : Color.light.whiteText;

  return (
    <>
      {isTasksEmpty ? (
        <EmptyBucket
          prio={prio}
          backgroundColor={backgroundColor}
          textColor={textColor}
        />
      ) : (
        <ActiveBucket
          prio={prio}
          tasks={tasks}
          backgroundColor={backgroundColor}
          textColor={textColor}
        />
      )}
    </>
  );
};

export default TaskBucketByPrio;
