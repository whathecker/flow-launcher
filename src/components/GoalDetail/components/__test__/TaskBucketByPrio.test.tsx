import * as React from "react";
import renderer from "react-test-renderer";
import TaskBucketByPrio from "../TaskBucketByPrio";
import { Task } from "../../../../types/core/entity";
import { taskFilters } from "../../../../utils";

const tasks: Task[] = [
  {
    _id: "622605f8766c2b210fcc5890",
    description: "",
    goal_id: "61d024a2b4cda2847a224bdf",
    priority: {
      importance: "yes",
      tier: "highest",
      urgency: "yes",
    },
    status: "open",
    title: "Test1",
  },
  {
    _id: "622605fd766c2b210fcc5891",
    description: "",
    goal_id: "61d024a2b4cda2847a224bdf",
    priority: {
      importance: "yes",
      tier: "highest",
      urgency: "yes",
    },
    status: "open",
    title: "Test2",
  },
  {
    _id: "62260602766c2b210fcc5892",
    description: "",
    goal_id: "61d024a2b4cda2847a224bdf",
    priority: {
      importance: "yes",
      tier: "high",
      urgency: "no",
    },
    status: "open",
    title: "test3",
  },
  {
    _id: "62260606766c2b210fcc5893",
    description: "",
    goal_id: "61d024a2b4cda2847a224bdf",
    priority: {
      importance: "yes",
      tier: "high",
      urgency: "no",
    },
    status: "open",
    title: "Test4",
  },
  {
    _id: "62260609766c2b210fcc5894",
    description: "",
    goal_id: "61d024a2b4cda2847a224bdf",
    priority: {
      importance: "no",
      tier: "low",
      urgency: "no",
    },
    status: "open",
    title: "test5",
  },
];
const result = taskFilters.filterByPriorityScheme(tasks);

describe("Test TaskBucketByPrio component", () => {
  it("Should renders TaskBucketByPrio correctly", () => {
    const tree = renderer
      .create(
        <TaskBucketByPrio
          prio={"low"}
          tasks={result.low}
          goalColor={"#F0B541"}
        />,
      )
      .toJSON();
    const tree2 = renderer
      .create(
        <TaskBucketByPrio
          prio={"mid"}
          tasks={result.mid}
          goalColor={"#F0B541"}
        />,
      )
      .toJSON();
    const tree3 = renderer
      .create(
        <TaskBucketByPrio
          prio={"high"}
          tasks={result.high}
          goalColor={"#F0B541"}
        />,
      )
      .toJSON();
    const tree4 = renderer
      .create(
        <TaskBucketByPrio
          prio={"highest"}
          tasks={result.highest}
          goalColor={"#F0B541"}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
    expect(tree3).toMatchSnapshot();
    expect(tree4).toMatchSnapshot();
  });
});
