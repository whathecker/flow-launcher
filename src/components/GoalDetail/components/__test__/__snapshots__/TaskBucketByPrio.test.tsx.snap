// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TaskBucketByPrio component Should renders TaskBucketByPrio correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#848484",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "height": 120,
      "marginBottom": "3%",
      "marginTop": 5,
      "opacity": 1,
      "paddingBottom": "8%",
      "paddingTop": "4%",
      "shadowColor": "#171717",
      "shadowOffset": Object {
        "height": 3,
        "width": 3,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingLeft": 15,
          "width": "95%",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-SemiBold",
            "fontSize": 15,
            "paddingLeft": 18,
            "paddingTop": 5,
          },
        ]
      }
    >
      Low priority
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "paddingBottom": "2%",
          "paddingLeft": "10%",
          "paddingTop": "6%",
          "width": "95%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 13,
          },
        ]
      }
    >
      test5
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "position": "absolute",
            "right": 5,
            "top": 45,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Test TaskBucketByPrio component Should renders TaskBucketByPrio correctly 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FDFCFC",
      },
      Object {
        "borderColor": "#848484",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "height": 120,
        "marginBottom": "3%",
        "marginTop": 5,
        "paddingBottom": "8%",
        "paddingTop": "4%",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FDFCFC",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingLeft": 15,
          "width": "95%",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 28,
          "width": 28,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#848484",
          },
          Object {
            "fontFamily": "Lexend-SemiBold",
            "fontSize": 15,
            "paddingLeft": 22,
            "paddingTop": 5,
          },
        ]
      }
    >
      Mid priority
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FDFCFC",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingTop": "9.5%",
          "width": "95%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#848484",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 12,
          },
        ]
      }
    >
      No tasks in this list
    </Text>
  </View>
</View>
`;

exports[`Test TaskBucketByPrio component Should renders TaskBucketByPrio correctly 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#848484",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "height": 120,
      "marginBottom": "3%",
      "marginTop": 5,
      "opacity": 1,
      "paddingBottom": "8%",
      "paddingTop": "4%",
      "shadowColor": "#171717",
      "shadowOffset": Object {
        "height": 3,
        "width": 3,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingLeft": 15,
          "width": "95%",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-SemiBold",
            "fontSize": 15,
            "paddingLeft": 18,
            "paddingTop": 5,
          },
        ]
      }
    >
      High priority
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "paddingBottom": "2%",
          "paddingLeft": "10%",
          "paddingTop": "6%",
          "width": "95%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 13,
          },
        ]
      }
    >
      test3
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 13,
            "paddingTop": 5,
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      + 1 more
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "position": "absolute",
            "right": 5,
            "top": 45,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Test TaskBucketByPrio component Should renders TaskBucketByPrio correctly 4`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#848484",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "height": 120,
      "marginBottom": "3%",
      "marginTop": 5,
      "opacity": 1,
      "paddingBottom": "8%",
      "paddingTop": "4%",
      "shadowColor": "#171717",
      "shadowOffset": Object {
        "height": 3,
        "width": 3,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingLeft": 15,
          "width": "95%",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-SemiBold",
            "fontSize": 15,
            "paddingLeft": 18,
            "paddingTop": 5,
          },
        ]
      }
    >
      Highest priority
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "paddingBottom": "2%",
          "paddingLeft": "10%",
          "paddingTop": "6%",
          "width": "95%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 13,
          },
        ]
      }
    >
      Test1
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 13,
            "paddingTop": 5,
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      + 1 more
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "position": "absolute",
            "right": 5,
            "top": 45,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      />
    </View>
  </View>
</View>
`;
